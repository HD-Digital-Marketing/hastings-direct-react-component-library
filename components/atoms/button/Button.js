import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './button.scss'

const Button = props => {

    const {
        type = 'button',
        url = '',
        className = '',
        theme = '',
        children,
        onClick
    } = props

    const buttonClass = classNames('a-btn', className, {
        'a-btn--primary': theme === 'primary',
        'a-btn--secondary': theme === 'secondary'
    })

    const isAnchor = url && (url.includes('http') || url.startsWith('#') || url.startsWith('mailto') || url.startsWith('/'))

    const renderAnchor = () => {
        <a href={url} className={buttonClass}>{children}</a>
    }

    const renderButton = () => {
        <button {...{ type, onClick }} className={buttonClass}>{children}</button>
    }

    return (
        isAnchor ? renderAnchor() : renderButton()
    )
}

Button.propTypes = {
  type: PropTypes.string,
  url: PropTypes.string,
  className: PropTypes.string,
  theme: PropTypes.string,
  children: PropTypes.string,
  onClick: PropTypes.string
}

export default Button
