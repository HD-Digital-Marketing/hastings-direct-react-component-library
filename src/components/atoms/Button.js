import React from 'react';

const Button = (props) => {

    const {
        children,
        type = 'button'
    } = props;

    return(
        <button type={type}>{children}</button>
    )

}

export default Button;