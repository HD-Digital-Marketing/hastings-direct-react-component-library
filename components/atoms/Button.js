import React from 'react';
import '../../scss/atoms/button.scss'

const Button = (props) => {

    const {
        children,
        type = 'button'
    } = props;

    return(
        <button className="a-btn" type={type}>{children}</button>
    )

}

export default Button;