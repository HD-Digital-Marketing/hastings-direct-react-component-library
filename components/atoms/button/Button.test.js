import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Check the button renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create(<Button type="button" theme="primary">Click me I'm a button</Button>)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });
});