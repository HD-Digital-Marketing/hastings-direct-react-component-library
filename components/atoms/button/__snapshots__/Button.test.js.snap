// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the button renders it should match the snapshot 1`] = `
<button
  className="a-btn a-btn--primary"
  type="button"
>
  Click me I'm a button
</button>
`;
